e2e/test/iothub/service/RegistryManagerE2ETests.cs

```diff
@@ -42,24 +41,42 @@ public async Task RegistryManager_BadProxy_ThrowsException()
         [LoggedTestMethod]
         public async Task RegistryManager_AddAndRemoveDeviceWithScope()
         {
-            var registryManager = RegistryManager.CreateFromConnectionString(Configuration.IoTHub.ConnectionString);
-
+            string edgeId1 = _devicePrefix + Guid.NewGuid();
+            string edgeId2 = _devicePrefix + Guid.NewGuid();
             string deviceId = _devicePrefix + Guid.NewGuid();
 
-            var edgeDevice = new Device(deviceId)
+            using var registryManager = RegistryManager.CreateFromConnectionString(Configuration.IoTHub.ConnectionString);
+
+            try
             {
-                Capabilities = new DeviceCapabilities { IotEdge = true }
-            };
-            edgeDevice = await registryManager.AddDeviceAsync(edgeDevice).ConfigureAwait(false);
+                var edgeDevice1 = new Device(edgeId1)
+                {
+                    Capabilities = new DeviceCapabilities { IotEdge = true }
+                };
+                edgeDevice1 = await registryManager.AddDeviceAsync(edgeDevice1).ConfigureAwait(false);
 
-            var leafDevice = new Device(Guid.NewGuid().ToString()) { Scope = edgeDevice.Scope };
-            Device receivedDevice = await registryManager.AddDeviceAsync(leafDevice).ConfigureAwait(false);
+                var edgeDevice2 = new Device(edgeId2)
+                {
+                    Capabilities = new DeviceCapabilities { IotEdge = true },
+                    ParentScopes = { edgeDevice1.Scope },
+                };
+                edgeDevice2 = await registryManager.AddDeviceAsync(edgeDevice2).ConfigureAwait(false);
+
+                var leafDevice = new Device(deviceId) { Scope = edgeDevice2.Scope };
+                leafDevice = await registryManager.AddDeviceAsync(leafDevice).ConfigureAwait(false);
 
-            Assert.IsNotNull(receivedDevice);
-            Assert.AreEqual(leafDevice.Id, receivedDevice.Id, $"Expected Device ID={leafDevice.Id}; Actual Device ID={receivedDevice.Id}");
-            Assert.AreEqual(leafDevice.Scope, receivedDevice.Scope, $"Expected Device Scope={leafDevice.Scope}; Actual Device Scope={receivedDevice.Scope}");
-            await registryManager.RemoveDeviceAsync(leafDevice.Id).ConfigureAwait(false);
-            await registryManager.RemoveDeviceAsync(edgeDevice.Id).ConfigureAwait(false);
+                edgeDevice2.ParentScopes.FirstOrDefault().Should().Be(edgeDevice1.Scope, "The parent scope should be respected as set.");
+
+                leafDevice.Id.Should().Be(deviceId, "The device Id shoudl be respected as set.");
+                leafDevice.Scope.Should().Be(edgeDevice2.Scope, "The parent scope should be respected as set.");
```

Comment:
Shouldn't it be "The device scope should be respected as set"?

Reply:
totally. Thanks for catching.